### PR TITLE
Before social login redirect

### PR DESCRIPTION
**Pull request by rochamarcelo**

### CODE REVIEW

**Comment by rochamarcelo** [comment]
Related to https://github.com/CakeDC/users/pull/782

File: composer.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
         "cakedc/auth": "^3.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^6",
+        "phpunit/phpunit": "5.7.*",
```
Comment (steinkel) [inline composer.json]: use | syntax instead? like the core does

File: src/Auth/SocialAuthenticate.php [hunk 2]
```diff
@@ -284,10 +284,11 @@ public function unauthenticated(ServerRequest $request, Response $response)
         }
 
         $authParams = $this->getConfig(sprintf('providers.%s.authParams', $request->getParam('provider')), []);
+        $location = $provider->getAuthorizationUrl($authParams);
 
-        $response = $response->withLocation($provider->getAuthorizationUrl($authParams));
+        $this->dispatchEvent(UsersAuthComponent::EVENT_BEFORE_SOCIAL_LOGIN_REDIRECT, compact('location', 'request'));
```
Comment (steinkel) [inline src/Auth/SocialAuthenticate.php]: event could return another location for example?

File: src/Controller/Traits/LinkSocialTrait.php [hunk 3]
```diff
@@ -40,6 +41,10 @@ public function linkSocial($alias = null)
             $this->request->getSession()->write('temporary_credentials', $temporaryCredentials);
         }
         $authUrl = $provider->getAuthorizationUrl($temporaryCredentials);
+        $this->dispatchEvent(UsersAuthComponent::EVENT_BEFORE_SOCIAL_LOGIN_REDIRECT, [
```
Comment (steinkel) [inline src/Controller/Traits/LinkSocialTrait.php]: what does the event could return?

**Pull request merged**